lib/metadata_fields.rb

```diff
@@ -0,0 +1,215 @@
+module MetadataFields
+	extend ActiveSupport::Concern
+	extend self
+
+	NAMESPACE_ALL = 'all'
+	MULTIPLE_ALL  = 'all'
+	DEFAULT_NAMESPACE = NAMESPACE_ALL
+	DEFAULT_MULTIPLE  = MULTIPLE_ALL
+	METADATA_FIELDS = YAML.load_file(File.join(Rails.root, "config", "metadata_fields.yml"))["terms"]["vocabulary"].deep_symbolize_keys
+	DISPLAY_CONFIG = YAML.load_file(File.join(Rails.root, "config", "metadata_fields.yml"))["config"]
+
+	# get metadata fields
+	def process_metadata_field_names(ns:DEFAULT_NAMESPACE, multiple: DEFAULT_MULTIPLE)
+		unless allowed_values_for_ns.include?(ns)
+			raise ArgumentError.new("#{ns} should be one of these values: #{allowed_values_for_ns}")
+		end
+		unless allowed_values_for_multiple.include?(multiple)
+			raise ArgumentError.new("#{multiple} should be one of these values: #{allowed_values_for_multiple}")
+		end
+		chk_key = ns.to_sym
+		field_names = []
+		if METADATA_FIELDS.has_key?(chk_key)
+			add_field_names_by_ns(multiple, field_names, chk_key)
+		elsif ns == NAMESPACE_ALL
+			add_all_field_names(multiple, field_names)
+		end
+		field_names
+	end
+
+	# get namespace and uri for
+	# metadata namespace
+	def get_source_info(ns:DEFAULT_NAMESPACE)
+		unless allowed_values_for_ns.include?(ns)
+			raise ArgumentError.new("#{ns} should be one of these values: #{allowed_values_for_ns}")
+		end
+		chk_key = ns.to_sym
+		if METADATA_FIELDS.has_key?(chk_key)
+			get_metadata_source_info(chk_key)
+		elsif ns == NAMESPACE_ALL
+			get_all_sources_info
+		end
```

Comment:
@jgpawletko @chrpr ,

I think I wrote it this way because I have an [unless](https://github.com/NYULibraries/ichabod/blob/chore/113569497_full-display-runs-off-yaml/lib/metadata_fields.rb#L33-L35) statement that traps the namespace value if it doesn't meet the expected parameters. So nil should not be returned because the if/elseif statement checks to see if either the ns is a key in the metadata hash(which is taken from the yaml file) or is the default value which is declared as a constant.